docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +27,180 @@ dependencies {
+
+[source,java]
+----
+@SpringBootApplication
+public class SpannerTemplateExample {
+
+	@Autowired
+	SpannerOperations spannerTemplate;
+
+  public void doSomething() {
+			this.spannerOperations.delete(Trade.class, KeySet.all());
+      //...
+			Trade t = new Trade();
+      //...
+			this.spannerOperations.insert(t);
+      //...
+			List<Trade> tradesByAction = spannerOperations.findAll(Trade.class);
+      //...
+	}
+}
+----
+
+The Template API provides convenience methods for:
+
+- https://cloud.google.com/spanner/docs/reads[Reads], and by providing SpannerReadOptions and SpannerQueryOptions
+   ** Stale read
+   ** Read with secondary indices
+   ** Read with limits and flow control
+- https://cloud.google.com/spanner/docs/reads#execute_a_query[Queries]
+- DML operations (delete, insert, update, upsert)
+- partial reads: you can define a set of columns to be read into your entity
+- partial writes: if you have only a few properties that you want to persist from your entity, you can select those
+
+
+=== Supported Types
+
+Spring Data Spanner supports the following types for regular fields:
+
+* `com.google.cloud.ByteArray`
+* `com.google.cloud.Date`
+* `com.google.cloud.Timestamp`
+* `java.lang.Boolean`
+* `java.lang.Long`
+* `java.lang.String`
+* `double[]`
+* `long[]`
+* `boolean[]`
+
+Spring Data Spanner supports the following inner types for `List` fields:
+
+* `com.google.cloud.ByteArray`
+* `com.google.cloud.Date`
+* `com.google.cloud.Timestamp`
+* `java.lang.Boolean`
+* `java.lang.Long`
+* `java.lang.String`
+
+
+=== Repositories
+
+{spring-data-commons-ref}/#repositories[Spring Data Repositories] are a powerful abstraction that can save you a lot of typing.
+For Spring Data Spanner, the code inheriting `SpannerRepository` gets all the benefits of `CrudRepository` and `PagingAndSortingRepository` as well, meaning you get by default a lot of generated methods, ready to use.
+
+For example:
 
 [source,java]
 ----
-@Repository
-public interface StudentRepository extends PagingAndSortingRepository<SimpleStudent, Key> {
+public interface TradeRepository extends SpannerRepository<Trade> {
 
-	List<SimpleStudent> findTop3DistinctByIdAndNameOrAgeOrderByAgeDesc(
-			String id, String name, int age);
+	List<Trade> findByAction(String action);
 
-	int countByAgeLessThanEqual(int age);
+	int countByAction(String action);
+
+  //named method are powerful but can get unwieldy
+	List<Trade> findTop3DistinctByActionAndSymbolOrTraderIdOrderBySymbolDesc(
+  			String action, String symbol, String traderId);
 
 	// This method uses the query from the properties file instead of one generated based on name.
-	List<SimpleStudent> fetchBySomeQuery(String someArg);
-}
+	List<Trade> fetchByActionNamedQuery(String action);
 
+}
 ----
 
-These can be used in an application like this:
+This repository can be used without an actual implementation!
```

Comment:
Remove the `!`

Reply:
but I'm enthusiastic about it, because it's cool! :) okay, I'll remove it.